### PR TITLE
feat: switch `alias` to `json`

### LINKED ISSUE #908: Consider switching "alias" annotation to "json"
The rationale is twofold:

1. It is almost certain that if a type is to be shared across multiple ingress methods, such as HTTP, gRPC, and GraphQL, users will want to be able to use different aliases for different protocols. The choice of `alias` doesn't allow for that.
2. It will simplify interacting with HTTP ingress endpoints from Go code such as integration tests, because their will only need to be a single annotation, `json:"xyz"`.

Schema should change to:

```
data Foo {
  field String alias http bar
}

verb bar() 
  ingress http GET /bar
```

Go annotations should change to:

```
json:"bar"
```

Kotlin annotation should change to something like this, but dig into whether we should just use `@JsonProperty`? Is this part of JAX-RS or some random thing from Jackson?

```
@JsonProperty("bar")
```

Edit: there doesn't seem to be any standard at all. Gson uses `SerializedName`, Jackson uses `@JsonProperty` - let's just use `@Json`.

### PR DESCRIPTION
**Pull request by wesbillman**
Fixes #908

### CODE REVIEW

File: backend/schema/field.go [hunk 1]
```diff
@@ -12,24 +12,24 @@ import (
...
 	Pos Position `parser:"" protobuf:"1,optional"`
 
-	Comments []string `parser:"@Comment*" protobuf:"3"`
-	Name     string   `parser:"@Ident" protobuf:"2"`
-	Type     Type     `parser:"@@" protobuf:"4"`
-	Alias    string   `parser:"('alias' @Ident)?" protobuf:"5"`
+	Comments  []string `parser:"@Comment*" protobuf:"3"`
+	Name      string   `parser:"@Ident" protobuf:"2"`
+	Type      Type     `parser:"@@" protobuf:"4"`
+	HTTPAlias string   `parser:"('alias' 'http' @Ident)?" protobuf:"5"`
```
Comment (wesbillman) [inline backend/schema/field.go:18]: Just want to check that we're ok with this new `HTTPAlias` name.

Comment (wesbillman) [reply backend/schema/field.go:18]: Also, @alecthomas does this parser change look correct to you? I sort of just guessed 😬

Comment (alecthomas) [reply backend/schema/field.go:18]: Looks good!

File: backend/controller/ingress/alias_test.go [hunk 2]
```diff
@@ -12,11 +12,11 @@ func TestTransformFromAliasedFields(t *testing.T) {
 	schemaText := `
 		module test {
 			data Inner {
-				waz String alias foo
+				waz String alias http foo
```
Comment (alecthomas) [inline backend/controller/ingress/alias_test.go:15]: I'm wondering if we should call the aliasing `json`? It is currently specific to the http ingress, but it is also only to do with JSON fields, not HTTP in general. I've gone back and forth on it, WDYT @wesbillman @worstell ?

Just thinking about it again - it's called "JSON" in each language. Let's call it that here too.

**Review by worstell — APPROVED** [review]

**Review by alecthomas — APPROVED** [review]

File: backend/controller/ingress/alias_test.go (hunk 2 above)
Comment (wesbillman) [reply backend/controller/ingress/alias_test.go:15]: yeah I like that too!

**Pull request merged**